### DISCUSSION TITLE
Multi-thread

### DISCUSSION THREAD

**Discussion by ertankucukoglu** [HashLoad/horse#157] [Q&A] [answered]
Hello,

I just learned about Horse. It seems really simple to use and easy to read its code. Middleware codes are mostly one line to include in. Thanks for making it a MIT license.

Most of the REST servers need database connection in a way. I see sample YouTube videos exampling to use mongo DB. I mostly need to use other databases (SQL Server, FirebirdSQL, etc) as to project needs. 

I just cannot be sure if Horse procedures for GET/POST requests runs in a separate thread or in main thread. Would you clarify that, please?

Depending on the thread they run, I need to create a separate database connection (probably use database connection pool) for each request or use one single connection for the whole project.

Thanks & Regards,
Ertan

**Comment by viniciussanchez** [comment, marked as answer]
Horse is multi thread
